File: src/System.Windows.Forms/tests/IntegrationTests/ScratchProject/TestDatas.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
using System.Runtime.Serialization;

namespace ScratchProject;

[Serializable]
internal sealed class TestData
{
    public TestData(string text1, string text2)
    {
        _text1 = text1;
        _text2 = text2;
    }

    public string _text1;
    public string _text2;
}

[Serializable]
internal sealed class TestData1
{
    public TestData1(PointF point)
    {
        _point = point;
    }

    public PointF _point;
}

[Serializable]
internal sealed class TestData2
{
    public TestData2(ImageListStreamer streamer)
    {
        _streamer = streamer;
    }

    public ImageListStreamer _streamer;
}

[Serializable]
internal sealed class TestData3
{
    public TestData3(Bitmap bitmap)
    {
        _bitmap = bitmap;
    }

    public Bitmap _bitmap;

    internal class InnerData
    {
        public InnerData(string text)
        {
            _text = text;
        }

        public string _text;
    }
}

[Serializable]
public sealed class TestData4 : ISerializable
{
    private readonly int _n1;

    public TestData4()
    {
        _n1 = -1;
    }

    private TestData4(SerializationInfo info, StreamingContext context)
    {
        _n1 = (info.GetValue(nameof(_n1), typeof(int)) is object)
            ? (int)info.GetValue(nameof(_n1), typeof(int))!
            : -1;
    }

    // The following method serializes the instance.
    void ISerializable.GetObjectData(SerializationInfo info, StreamingContext context)
    {
        info.AddValue(nameof(_n1), _n1);
    }
}

    internal class BinaryFormatReadableData : IEquatable<BinaryFormatReadableData>, IDisposable
    {
        public BinaryFormatReadableData()
        {
            _point = new PointF(1.2F, 1.2F);
            //_bitmap = new(10, 10);
            //_imageList = new();
            _id = ++s_counter;
        }

        // public Bitmap _bitmap;
        public PointF _point;
        // private readonly ImageList _imageList;
        private readonly int _id;
        private static int s_counter;

        public bool Equals(BinaryFormatReadableData? other)
            => other is object && _point == other._point /* && _bitmap.Size == other._bitmap.Size */ && _id == other._id;
        public override bool Equals(object? obj) => obj is BinaryFormatReadableData other && Equals(other);
        public override int GetHashCode() => _point.GetHashCode();
        public void Dispose()
        {
            //_bitmap?.Dispose();
            //_imageList?.Dispose();
        }
    }

File: src/System.Windows.Forms/tests/IntegrationTests/ScratchVBProject/TestDatas.vb
' Licensed to the .NET Foundation under one or more agreements.
' The .NET Foundation licenses this file to you under the MIT license.
Imports System.Runtime.Serialization

<Serializable>
Friend NotInheritable Class TestData
    Public Sub New(text1 As String, text2 As String)
        _text1 = text1
        _text2 = text2
    End Sub

    Public _text1 As String
    Public _text2 As String
End Class

<Serializable>
Friend NotInheritable Class TestData1
    Public Sub New(point As PointF)
        _point = point
    End Sub

    Public _point As PointF
End Class

<Serializable>
Friend NotInheritable Class TestData2
    Public Sub New(streamer As ImageListStreamer)
        _streamer = streamer
    End Sub

    Public _streamer As ImageListStreamer
End Class

<Serializable>
Friend NotInheritable Class TestData3
    Public Sub New(bitmap As Bitmap)
        _bitmap = bitmap
    End Sub

    Public _bitmap As Bitmap

    Friend NotInheritable Class InnerData
        Public Sub New(text As String)
            _text = text
        End Sub

        Public _text As String
    End Class
End Class

<Serializable>
Public NotInheritable Class TestData4
    Implements ISerializable

    Private ReadOnly _n1 As Integer

    Public Sub New()
        _n1 = -1
    End Sub

    Private Sub New(info As SerializationInfo, context As StreamingContext)
        _n1 = If(info.GetValue(NameOf(_n1), GetType(Integer)) IsNot Nothing, CInt(info.GetValue(NameOf(_n1), GetType(Integer))), -1)
    End Sub

    ' The following method serializes the instance.
#Disable Warning SYSLIB0050 ' Type or member is obsolete
    Private Sub GetObjectData(info As SerializationInfo, context As StreamingContext) Implements ISerializable.GetObjectData
#Enable Warning SYSLIB0050 ' Type or member is obsolete
        info.AddValue(NameOf(_n1), _n1)
    End Sub
End Class

